### PR TITLE
Apply More Common::BitSize and CHAR_BIT

### PR DESCRIPTION
**Pull request by Minty-Meeo**
Tiny refactor.  I could go further in a few places, but didn't feel like speculating too hard.  These are the obvious ones.

### CODE REVIEW

**Comment by Minty-Meeo** [comment]
Perhaps building my PR before submitting it would be wise.

**Comment by Minty-Meeo** [comment]
Looks like I need to update my clang-format.  It disagrees on the formatting of the macro in Zelda.cpp.

File: Source/Core/Core/PowerPC/Jit64Common/EmuCodeBlock.cpp [hunk 1]
```diff
@@ -215,14 +216,14 @@ class MMIOReadCodeGenerator : public MMIO::ReadHandlingMethodVisitor<T>
   {
   }
 
-  void VisitConstant(T value) override { LoadConstantToReg(8 * sizeof(T), value); }
+  void VisitConstant(T value) override { LoadConstantToReg(CHAR_BIT * sizeof(T), value); }
```
Comment (BhaaLseN) [inline Source/Core/Core/PowerPC/Jit64Common/EmuCodeBlock.cpp:219]: Any particular reason why you didn't use `BitSize<T>()` instead? It seems to fit the same pattern as the others, at least on ~~paper~~ GitHub.

Comment (Minty-Meeo) [reply Source/Core/Core/PowerPC/Jit64Common/EmuCodeBlock.cpp:219]: I did at first, but this group of functions accept the bit size parameter as an int for reasons beyond my control / outside the scope of this PR.  This caused made MSVC throw a warning turned error about potential loss of information in converting from size_t to int.  This was the next best solution, then.

**Comment by Minty-Meeo** [comment]
Can I get an update on this PR?

**Review by phire — APPROVED** [review]

**Comment by Minty-Meeo** [comment]
Came up with an improvement to Common::BitSize thanks to NeoBrain pointing me towards std::integral_constant.  Decided to tack it onto this PR which was just kinda... treading water I guess?
Doing it this way doesn't change any syntax, though I wonder if it would be better if it did?

**Comment by Minty-Meeo** [comment]
So then here is what BitSize can look like as an inline constexpr variable.  Getting rid of the parenthesis looks good, imo.

**Comment by Minty-Meeo** [comment]
Or, option 3.  Take the original function and upgrade it to be consteval now that we are in C++20.

**Comment by Minty-Meeo** [comment]
imo I prefer the templated inline constexpr variable version, but I know the upgrade to consteval is the simplest to justify.

**Comment by Minty-Meeo** [comment]
Also, side rant.  I don't understand how consteval variables are not a thing yet.  I think there ought to be a kind of constant expression (not to be confused with constexpr) which absolutely never can end up existing at runtime to have its address taken.

**Comment by Minty-Meeo** [comment]
Reverting to constexpr due to buildbots.

**Pull request closed without merging**